src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -171,6 +171,34 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
     }
     
     /**
+     * Generates the feedback session closed emails for the given {@code session}.
+     * @throws EntityDoesNotExistException
+     */
+    public List<EmailWrapper> generateFeedbackSessionClosedEmails(FeedbackSessionAttributes session) {
+        
+        if (session.isPrivateSession()) {
+            return new ArrayList<EmailWrapper>();
+        }
+        
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        boolean isEmailNeededForStudents = false;
+        try {
+            isEmailNeededForStudents = fsLogic.isFeedbackSessionHasQuestionForStudents(
+                    session.getFeedbackSessionName(), session.getCourseId());
+        } catch (EntityDoesNotExistException e) {
+            log.severe("Course " + session.getCourseId() + " does not exist or "
+                    + "session " + session.getFeedbackSessionName() + " does not exist");
+        }
+        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(session.getCourseId());
+        if (isEmailNeededForStudents) {
+            students = studentsLogic.getStudentsForCourse(session.getCourseId());
+        }
```

Comment:
Now this can be:
```java
List<StudentAttributes> students = isEmailNeededForStudents
                                   ? ...
                                   : ...;
```